File: packages/client-python/src/rocketride/types/deploy.py
# MIT License
#
# Copyright (c) 2026 Aparavi Software AG
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

"""
Deploy type definitions for the RocketRide Python SDK.

Mirrors the server's ``DeploymentRecord`` model returned by ``rrext_deploy_*`` commands.

Types:
    DeploymentRecord:   Deployment record as returned by add / list / status.
"""

from typing import Literal, TypedDict


class DeploymentRecord(TypedDict, total=False):
    """
    Deployment record returned by ``rrext_deploy_add``, ``rrext_deploy_list``,
    and ``rrext_deploy_status``.

    Mirrors ``DeploymentRecord.model_dump()`` on the server.
    """

    deployment_id: str
    name: str
    pipeline: dict
    schedule: str
    state: Literal['active', 'paused', 'errored']
    created_by: str
    created_at: float
    updated_at: float

File: packages/client-typescript/src/client/types/deploy.ts
/**
 * MIT License
 *
 * Copyright (c) 2026 Aparavi Software AG
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

/**
 * Deploy type definitions for the RocketRide TypeScript SDK.
 *
 * Mirrors the server's `DeploymentRecord` model returned by `rrext_deploy_*` commands.
 */

// =============================================================================
// DEPLOY TYPES
// =============================================================================

/** Deployment record returned by `rrext_deploy_add`, `rrext_deploy_list`, and `rrext_deploy_status`. */
export interface DeploymentRecord {
	deployment_id?: string;
	name?: string;
	pipeline?: Record<string, unknown>;
	/** Cron expression or `"manual"`. */
	schedule?: string;
	state?: 'active' | 'paused' | 'errored';
	created_by?: string;
	/** Unix timestamp (seconds). */
	created_at?: number;
	/** Unix timestamp (seconds). */
	updated_at?: number;
}

/** Parameters accepted by `client.deploy.update()`. */
export interface DeployUpdateParams {
	pipeline?: Record<string, unknown>;
	schedule?: string;
}
